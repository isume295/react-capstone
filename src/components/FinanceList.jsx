import React, { useEffect, useState } from 'react';
import { styled } from 'styled-components';
import { useDispatch, useSelector } from 'react-redux';
import FinancialData from './FinancialData';
import { getFinance } from '../redux/finance/FinanceSlice';

export default function FinanceList() {
  const { finance, isLoading } = useSelector((store) => store.finance);
  const [searchInput, setSearchInput] = useState('');

  const handleSearchInput = (event) => {
    setSearchInput(event.target.value);
  };

  const filteredCompanies = searchInput === ''
    ? finance
    : finance.filter((company) => company.name.toLowerCase().includes(searchInput.toLowerCase()));

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getFinance());
  }, []);

  return (
    <Div>
      {isLoading === true ? <h3>Loading</h3> : ''}
      <div className="search-container">
        <input type="text" onChange={handleSearchInput} value={searchInput} placeholder="search company" />
      </div>
      <DataContainer>
        {filteredCompanies.map((data) => (
          <FinancialData key={data.symbol} data={data} className={filteredCompanies.length % 2 === 0 ? 'even' : ''} />
        ))}
      </DataContainer>
    </Div>
  );
}

const Div = styled.div`
width: 100%;
display: flex;
flex-direction: column;
gap: 10px;
justify-content: center;
align-items: center;
background-color: #EC4C8A;

.search-container{
    margin-top: 5px;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
   
   input {
    width: 80%;
    padding: 7px 3px;
    border: 2px solid black;
    border-radius: 10px;
    outline: none;
   }
}
`;

const DataContainer = styled.div`
width: 100%;
display: grid;
grid-template-columns: auto auto;

`;
