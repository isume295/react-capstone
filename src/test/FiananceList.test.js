import React from 'react';
import renderer from 'react-test-renderer';
import FinanceList from './__mocks__/FinanceList';

describe('Finance List', () => {
  it('renders correctly', () => {
    const component = renderer.create(<FinanceList />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
