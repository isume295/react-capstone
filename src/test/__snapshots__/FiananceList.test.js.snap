// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Finance List renders correctly 1`] = `
<div>
  <div>
    <button
      className="more"
      type="button"
    >
      view more
    </button>
    <div
      className="details"
    >
      <p>
        symbol
      </p>
      <span>
        name
      </span>
    </div>
  </div>
</div>
`;
